### ISSUE TITLE
Symbol Conflicts b/c macro definitions

### ISSUE THREAD

**Issue by dad4x** [Tencent/rapidjson#1499]
An environment I'm working in #defines K and W, which messes up some things in diyfp.h and dtoa.h.   It would make life easier by changing

GetCachedPower( int e, int* K ) -> GetCachedPower( int e, int* K_ )

DigitGen( DiyFP& W ... int* K ) -> DigitGen( DiyFP& W_ ... int *K_ )

or some other alternate.

**Comment by h46incon** [comment]
I think you should not define macro with a very short name in headers.

**Comment by dad4x** [comment]
If you're working on a very large existing codebase, sometimes you don't get a choice and have to live with other people's historical misjudgements on such things.

**Comment by h46incon** [comment]
No, mistakes should be fixed. `K` is a very very normal variable name, and you will meet it in other projects.
